File: app/src/main/java/yellow7918/ajou/ac/aunager/login/SignInActivity.java
package yellow7918.ajou.ac.aunager.login;


import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.widget.Button;
import android.widget.EditText;

import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

import yellow7918.ajou.ac.aunager.BaseActivity;
import yellow7918.ajou.ac.aunager.MainActivity;
import yellow7918.ajou.ac.aunager.R;


public class SignInActivity extends BaseActivity {

    private FirebaseAuth firebaseAuth;
    private ProgressDialog progressDialog;

    private Button signInButton;
    private Button signUpButton;

    private String email;
    private String password;
    private FirebaseAuth auth;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_in);

        progressDialog = new ProgressDialog(this);
        firebaseAuth = FirebaseAuth.getInstance();

        if (firebaseAuth.getCurrentUser() != null) {
            Intent i = new Intent(SignInActivity.this, MainActivity.class);
            startActivity(i);
            finish();
        }

        EditText emailEdit = findViewById(R.id.email_edit);
        EditText passwordEdit = findViewById(R.id.password_edit);

        signInButton = findViewById(R.id.button_sign_in);
        signInButton.setOnClickListener(v -> {
            showProgressBar("잠시만 기다려 주세요.");

            email = emailEdit.getText().toString();
            password = passwordEdit.getText().toString();

            auth = FirebaseAuth.getInstance();

            if (email.length() <= 0) {
                emailEdit.setError("아이디를 입력해 주세요.");
                hideProgressBar();
            } else if (password.length() <= 0) {
                passwordEdit.setError("비밀번호를 입력해 주세요.");
                hideProgressBar();
            } else {
                Task<AuthResult> authResultTask = auth.signInWithEmailAndPassword(email, password);
                authResultTask.addOnSuccessListener(authResult -> {
                    hideProgressBar();
                    Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                    startActivity(intent);
                    finish();
                });
                authResultTask.addOnFailureListener(e -> {
                    hideProgressBar();
                    Snackbar.make(v, "로그인에 실패하였습니다.", Snackbar.LENGTH_SHORT).show();
                });
            }
        });

        signUpButton = findViewById(R.id.button_sign_up);
        signUpButton.setOnClickListener(view -> {
            Intent intent = new Intent(getBaseContext(), SignUpActivity.class);
            startActivity(intent);
        });

    }

    private void showProgressBar(String text) {
        progressDialog.setMessage(text);
        progressDialog.show();
        signInButton.setEnabled(false);
        signUpButton.setEnabled(false);
    }

    private void hideProgressBar() {
        progressDialog.dismiss();
        signInButton.setEnabled(true);
        signUpButton.setEnabled(true);
    }
}

File: app/src/main/java/yellow7918/ajou/ac/aunager/SignInActivity.kt
package yellow7918.ajou.ac.aunager

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.view.View
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import kotlinx.android.synthetic.main.activity_sign_in.*

class SignInActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sign_in)

        val auth: FirebaseAuth = FirebaseAuth.getInstance()


        auth.currentUser?.let {
            autoLogin()
        }


        button_sign_up.setOnClickListener {
            startActivity(Intent(this@SignInActivity, SignUpActivity::class.java))
        }

        button_sign_in.setOnClickListener { v ->
            button_sign_in.visibility = View.GONE
            val email = email_edit.text.toString()
            val password = password_edit.text.toString()

            auth.signInWithEmailAndPassword(email, password).addOnSuccessListener {
                startActivity(Intent(this@SignInActivity, MainActivity::class.java))
                finish()
            }.addOnFailureListener {
                Snackbar.make(v, it.localizedMessage, Snackbar.LENGTH_SHORT).show()
                button_sign_in.visibility = View.VISIBLE
            }
        }
    }

    private fun autoLogin(): Unit {
        Toast.makeText(this@SignInActivity, "Hello", Toast.LENGTH_SHORT).show()
        startActivity(Intent(this@SignInActivity, MainActivity::class.java))
        finish()
    }
}

/*
private fun FirebaseUser.autoLogin(activity: AppCompatActivity): Unit {
    Toast.makeText(activity, "Hello", Toast.LENGTH_SHORT).show()
    activity.startActivity(Intent(activity, MainActivity::class.java))
    activity.finish()
}
*/